### ISSUE TITLE
Bad dialing to peer unhandled

### ISSUE THREAD

**Issue by ChihChengLiang** [ethereum/trinity#1396]
### What is wrong?

```
  DEBUG  2019-12-13 13:00:20,004          TrioEndpoint  Endpoint[discovery]: starting client handler for <trio.SocketStream object at 0x10f7db9e8>
   DEBUG  2019-12-13 13:00:20,005        RemoteEndpoint  RemoteEndpoint connection established: discovery <-> bjson-rpc-api
   DEBUG  2019-12-13 13:00:20,005        RemoteEndpoint  RemoteEndpoint connection established: discovery <-> network-db
   DEBUG  2019-12-13 13:00:20,005        RemoteEndpoint  RemoteEndpoint connection established: bbeacon-node <-> discovery
   DEBUG  2019-12-13 13:00:20,005        RemoteEndpoint  RemoteEndpoint connection established: discovery <-> bbeacon-node
   DEBUG  2019-12-13 13:00:20,334                 swarm  dialed peer 16Uiu2HAmKg1NR2Y9GnVNRQX9LqGw8J9yaHmCNwGDsudStiJEtv8T over base transport
   DEBUG  2019-12-13 13:00:21,014                 swarm  fail to upgrade security for peer 16Uiu2HAmKg1NR2Y9GnVNRQX9LqGw8J9yaHmCNwGDsudStiJEtv8T
--- Logging error ---
   DEBUG  2019-12-13 13:00:21,014                  Node  Fail to dial peer_id: 16Uiu2HAmKg1NR2Y9GnVNRQX9LqGw8J9yaHmCNwGDsudStiJEtv8T, maddr: /ip4/54.235.12.61/tcp/30303/p2p/16Uiu2HAmKg1NR2Y9GnVNRQX9LqGw8J9yaHmCNwGDsudStiJEtv8T, error: fail to upgrade security for peer 16Uiu2HAmKg1NR2Y9GnVNRQX9LqGw8J9yaHmCNwGDsudStiJEtv8T
Traceback (most recent call last):
  File "/usr/local/var/pyenv/versions/3.7.0/envs/py-evm/lib/python3.7/site-packages/libp2p/transport/upgrader.py", line 40, in upgrade_security
    raw_conn, peer_id
  File "/usr/local/var/pyenv/versions/3.7.0/envs/py-evm/lib/python3.7/site-packages/libp2p/security/security_multistream.py", line 80, in secure_outbound
    secure_conn = await transport.secure_outbound(conn, peer_id)
  File "/usr/local/var/pyenv/versions/3.7.0/envs/py-evm/lib/python3.7/site-packages/libp2p/security/secio/transport.py", line 483, in secure_outbound
    local_nonce, local_peer, local_private_key, conn, peer_id
  File "/usr/local/var/pyenv/versions/3.7.0/envs/py-evm/lib/python3.7/site-packages/libp2p/security/secio/transport.py", line 424, in create_secure_session
    raise e
  File "/usr/local/var/pyenv/versions/3.7.0/envs/py-evm/lib/python3.7/site-packages/libp2p/security/secio/transport.py", line 420, in create_secure_session
    local_peer, local_private_key, remote_peer, msg_io, local_nonce
  File "/usr/local/var/pyenv/versions/3.7.0/envs/py-evm/lib/python3.7/site-packages/libp2p/security/secio/transport.py", line 311, in _establish_session_parameters
    "received_remote_peer": remote_peer_from_proposal,
libp2p.security.secio.exceptions.PeerMismatchException: {'expected_remote_peer': <libp2p.peer.id.ID (16Uiu2HAmKg1NR2Y9GnVNRQX9LqGw8J9yaHmCNwGDsudStiJEtv8T)>, 'received_remote_peer': <libp2p.peer.id.ID (16Uiu2HAmV1d47rkuDBvs4bx4pGgquUiLqes9UoBmiH7M6V93FUKw)>}

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/var/pyenv/versions/3.7.0/envs/py-evm/lib/python3.7/site-packages/libp2p/network/swarm.py", line 112, in dial_peer
    secured_conn = await self.upgrader.upgrade_security(raw_conn, peer_id, True)
  File "/usr/local/var/pyenv/versions/3.7.0/envs/py-evm/lib/python3.7/site-packages/libp2p/transport/upgrader.py", line 50, in upgrade_security
    ) from error
libp2p.transport.exceptions.SecurityUpgradeFailure: handshake failed when upgrading to secure connection

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/liangcc/projects/EAPL/eth2.0s/trinity/trinity/protocol/bcc_libp2p/node.py", line 359, in dial_peer_maddr
    addrs=[maddr],
  File "/usr/local/var/pyenv/versions/3.7.0/envs/py-evm/lib/python3.7/site-packages/libp2p/host/basic_host.py", line 156, in connect
    await self._network.dial_peer(peer_info.peer_id)
  File "/usr/local/var/pyenv/versions/3.7.0/envs/py-evm/lib/python3.7/site-packages/libp2p/network/swarm.py", line 117, in dial_peer
    raise SwarmException(error_msg % peer_id) from error
libp2p.network.exceptions.SwarmException: fail to upgrade security for peer 16Uiu2HAmKg1NR2Y9GnVNRQX9LqGw8J9yaHmCNwGDsudStiJEtv8T

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/liangcc/projects/EAPL/eth2.0s/trinity/trinity/protocol/bcc_libp2p/node.py", line 389, in dial_peer_maddr_with_retries
    await self.dial_peer_maddr(maddr, peer_id)
  File "/Users/liangcc/projects/EAPL/eth2.0s/trinity/trinity/protocol/bcc_libp2p/node.py", line 364, in dial_peer_maddr
    raise DialPeerError from e
trinity.protocol.bcc_libp2p.exceptions.DialPeerError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/liangcc/projects/EAPL/eth2.0s/trinity/trinity/_utils/logging.py", line 74, in emit
    msg_data = pickle.dumps(self.prepare(record))
  File "/Users/liangcc/projects/EAPL/eth2.0s/trinity/trinity/_utils/logging.py", line 63, in prepare
    msg = self.format(record)
  File "/usr/local/var/pyenv/versions/3.7.0/lib/python3.7/logging/__init__.py", line 829, in format
    return fmt.format(record)
  File "/usr/local/var/pyenv/versions/3.7.0/lib/python3.7/logging/__init__.py", line 569, in format
    record.message = record.getMessage()
  File "/usr/local/var/pyenv/versions/3.7.0/lib/python3.7/logging/__init__.py", line 331, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "<string>", line 1, in <module>
  File "/usr/local/var/pyenv/versions/3.7.0/lib/python3.7/multiprocessing/spawn.py", line 105, in spawn_main
    exitcode = _main(fd)
  File "/usr/local/var/pyenv/versions/3.7.0/lib/python3.7/multiprocessing/spawn.py", line 118, in _main
    return self._bootstrap()
  File "/usr/local/var/pyenv/versions/3.7.0/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
    self.run()
  File "/usr/local/var/pyenv/versions/3.7.0/lib/python3.7/multiproc
```

### How can it be fixed

Fill this in if you know how to fix it.

**Comment by NIC619** [comment]
Should be fixed by d77b5e5

**Comment by ChihChengLiang** [comment]
Closed by https://github.com/ethereum/trinity/commit/d77b5e5b6372d7af4d19bbf394c96b4d20e494b2 in #1288
